docs/pages/api-reference/operators.md

```diff
@@ -6,83 +6,104 @@ status: wip
+Operators in Fennel are functions that takes one or more datasets as inputs and return a dataset as output and are the
+building
+blocks of a pipeline. Then look very similar to the operators
+in [pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html).
 
-### Join ( Left Join )
+### Join
 
-Fennel allows you to join two datasets using the `left_join` operator. You can only join a dataset
+Fennel allows you to join two datasets using the `join` operator. You can only join a dataset
 against another dataset which has keys defined. The join operation is performed on the keys of the RHS dataset.
 
-The `left_join` operator has the following parameters:
+The `join` operator has the following parameters:
 
 1. `dataset: Dataset` - positional argument, that specifies the RHS Dataset.&#x20;
 2. `on: List[str]` - kwarg that specifies the list of fields to join on.
 3. `left_on: List[str]` - optional kwarg specifying the list of fields to join on for the LHS dataset.&#x20;
 4. `right_on: List[str]` - optional kwarg specifying the list of fields to join on for the RHS dataset.&#x20;
-5. `within: Tuple[Duration, Duration]` - optional kwarg specifying the time window within which the join should be performed.&#x20; 
-   1. The first value in the tuple represents how far back in time we can go, to perform a join. The term "forever" means that we can go infinitely back in time when searching for an event to join from the left-hand side (LHS) data. 
-   2. The second value in the tuple represents how far ahead in time we can go to perform a join. This is useful when you want to create a buffer duration where the RHS data of the join can come later.
-    
-    The default value for this parameter is `("forever", "0s")` which means that we can go infinitely back in time and the RHS data should be available for the event time of the LHS data.&#x20;
+5. `within: Tuple[Duration, Duration]` - optional kwarg specifying the time window within which the join should be
+   performed.&#x20;
+    1. The first value in the tuple represents how far back in time we can go, to perform a join. The term "forever"
+       means that we can go infinitely back in time when searching for an event to join from the left-hand side (LHS)
+       data.
+    2. The second value in the tuple represents how far ahead in time we can go to perform a join. This is useful when
+       you want to create a buffer duration where the RHS data of the join can come later.
+       The default value for this parameter is `("forever", "0s")` which means that we can go infinitely back in time
+       and
+       the RHS data should be available for the event time of the LHS data.&#x20;
+
+6. `how: str` - optional kwarg indicating whether the join should be an inner join (`how='inner'`) or a left-outer
+   join (`how='left'`). The default value is `'left'`.
```

Comment:
I may be forgetting but I think we talked about making 'how' as mandatory (so no default values)?

Reply:
you're right! Will change it to required.